### PR TITLE
Fix issue with transfer_balance in task pallet

### PR DESCRIPTION
**Pull request by vivekvpandya**

### CODE REVIEW

**Comment by stojanov-igor** [comment]
@vivekvpandya 

This looks good. Very clean.

So you didn't have to refactor the DispatchError in the rest of the pallets?
 
The tests work also. Nice job!

Just a small note: you can name the branches something related to the task. For example `feat/bugfix-task-transfer`. Branches are to be deleted after merging, but it should be descriptive to the task.

File: task/src/lib.rs [hunk 1]
```diff
@@ -407,19 +407,8 @@ pub mod pallet {
...
-			//let task_initiator = task.initiator.clone();
-			log::info!("budget {:?}.", budget);
-			log::info!("signer {:?}.", task_initiator);
-			log::info!("task_volunteer {:?}.", task_volunteer);
-			ensure!(<T as self::Config>::Currency::free_balance(task_initiator) >= budget, <Error<T>>::NotEnoughBalance);
-			let result = <T as self::Config>::Currency::transfer(task_initiator, task_volunteer, budget, ExistenceRequirement::KeepAlive);
-			log::info!("Result Output {:?}.", result);
-
-			Ok(())
+		pub fn transfer_balance(task_initiator: &T::AccountId, task_volunteer: &T::AccountId, budget: BalanceOf<T>) -> Result<(),  DispatchError> {
```
Comment (stojanov-igor) [inline task/src/lib.rs:411]: Claned up all the logs and comments.. Nice!

File: task/src/tests.rs [hunk 2]
```diff
@@ -368,11 +369,13 @@ fn decrease_task_count_when_removing_task(){
 }
 
 #[test]
-fn transfer_ballance_works(){
+fn transfer_balance_works(){
 	new_test_ext().execute_with( || {
 		
 		// Transfer balance works using Mock
+        // initially use has 10 units
```
Comment (stojanov-igor) [inline task/src/tests.rs:376]: I mock two accounts in the Genesis Config and each has 10 units. If you need, you can create more accounts and transfer more. Not necessary for now.

**Review by stojanov-igor — APPROVED** [review]

**Comment by vivekvpandya** [comment]
> So you didn't have to refactor the DispatchError in the rest of the pallets?

To make this test work, this is enough but I will update all usage, that is why I have marked it as draft PR.

File: task/src/lib.rs (hunk 1 above)
Comment (vivekvpandya) [reply task/src/lib.rs:411]: @JosephKnecht-lab `transfer_balance` does not do anything special then what `Currency::transfer` is doing. Is there any reason/purpose for this function which I am not able to see?

**Pull request merged**